core/src/main/kotlin/xtdb/indexer/LiveTable.kt

```diff
@@ -68,10 +85,10 @@ class LiveTable
         fun openWatermark(): Watermark = openWatermark(transientTrie)
         val docWriter: IVectorWriter = putWtr
         val liveRelation: IRelationWriter = this@LiveTable.liveRelation
+        val startPos = liveRelation.writerPosition().position
+        private var pos = startPos
```

Comment:
I'd be a little cautious here about representing the same mutable state in two different places - it adds the assumption that the two will remain in sync

(`val startPos` is fine, it's `var pos` that alerted my radar)